### PR TITLE
feature(highlight): enhancements and code-health.

### PR DESCRIPTION
**Pull request by emoralesb05**
# Description

Major enhancements, initial unit tests and XSS protection.

### What's included?

- **feature(highlight):** Added `[content]` input to load code content dynamically.
- **feature(highlight):** Support for any pre-built `highlight.js` theme.
- **feature(highlight):** Mimicking `VS Dark+` theme as closely as possible when importing `covalent-highlight-theme`.
- **bugfix(highlight):** Use `Renderer` to limit access to native DOM and `DomSanitizer` to prevent `XSS` issues. 
- **bugfix(highlight):** Remove only empty lines at the beginning and end of code content only to keep new lines within code.
- **bugfix(highlight):** 	Remove `html` workaround so code parsing is more generic now and `html` is properly displayed.
- **code-health(highlight):** Added initial set of unit-tests.
- Updated `README.md` and demo.
- Load `README.md` as resource in docs to avoid doc duplication.

#### Test Steps

- [x] `ng serve` or `ng serve --aot --prod`
- [x] Go to localhost:4200/#/components/markdown, see docs with loaded `README.md` and new default theme.
- [x] Compare with https://teradata.github.io/covalent/#/components/syntax-highlighting
- [x] `npm run test`
- [x] `npm run lint`
- [x] `npm run build`

##### Screenshots or link to CodePen/Plunker/JSfiddle

image

image

### CODE REVIEW

File: src/platform/highlight/highlight.component.spec.ts [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+    fixture.detectChanges();
+    fixture.whenStable().then(() => {
+      fixture.detectChanges();
+      expect(fixture.debugElement.query(By.css('td-highlight pre code'))).toBeTruthy();
+      expect(element.querySelector('td-highlight pre code').textContent.trim()).toContain(`{{property}}`);
+      expect(element.querySelectorAll('.hljs-tag').length).toBe(6);
+    });
+  })));
+
+  it('should render cpde from dynamic content', async(inject([], () => {
```
Comment (jeremysmartt) [inline src/platform/highlight/highlight.component.spec.ts]: typo: cpde

File: src/platform/markdown/README.md [hunk 2]
```diff
@@ -20,7 +20,7 @@ By default, `--dev` build will log the following message in the console to let y
 
 This component can be installed as npm package.
 
-```npm
+```bash
```
Comment (jeremysmartt) [inline src/platform/markdown/README.md]: this works on windows also?  should this say bash?

Comment (emoralesb05) [inline src/platform/markdown/README.md]: `bash` is the language used to apply color theme in `GitHub`

**Comment by jeremysmartt** [comment]
LGTM!

**Pull request merged**